src/compiler/tsbuildPublic.ts

```diff
@@ -212,6 +213,18 @@ namespace ts {
         originalGetSourceFile: CompilerHost["getSourceFile"];
     }
 
+    interface FileWatcherWithModifiedTime {
+        callbacks: FileWatcherCallback[];
+        watcher: FileWatcher;
+        modifiedTime: Date | undefined;

```

Comment:
This is undefined because we might not have queried the time for the file, eg tsbuildinfo is missing.